packages/fetchai/connections/oef/connection.py

```diff
@@ -427,12 +171,9 @@ def on_message(
         """
         # We are not using the 'msg_id', 'dialogue_id' and 'origin' parameters because 'content' contains a
         # serialized instance of 'Envelope', hence it already contains this information.
-        assert self.in_queue is not None
-        assert self.loop is not None
+        self._check_loop_and_queue()
         envelope = Envelope.decode(content)
-        asyncio.run_coroutine_threadsafe(
-            self.in_queue.put(envelope), self.loop
-        ).result()
+        asyncio.run_coroutine_threadsafe(self.in_queue.put(envelope), self.loop)
```

Comment:
do we not need to call result?

Reply:
the result will block and this result even not returned from the function